### ISSUE TITLE
Can't install @datorama/akita for Angular 15.2.8.

### ISSUE THREAD

**Issue by Igor2017** [salesforce/akita#1064]
### Is this a regression?

Yes

### Description

Cause: The package.json file specifies the version of "tslib": "2.4.1".

If you change the version to "tslib": "^2.4.1", then everything compiles, all tests pass and can be installed in Angular 15.2.8.

### Do you want to create a pull request?

Yes

**Comment by Igor2017** [comment]
The point is that the angular/cli command
`ng new app` creates a package.json in which
"tslib": "^2.3.0" which results in version "2.5.0" being installed.
You can solve the problem by replacing "tslib": "^2.3.0" in package.json with "tslib": "2.4.1".
But not everyone is up to it.

**Comment by simeyla** [comment]
Angular 16.01 has the following versions for a new app.

 "tslib": "^2.3.0"
 "typescript": "~5.0.2"

Why does Akita have a hardcoded version?
I don't want to replace ^2.3.0 with 2.4.1!

Is Akita going to be updated or is there some other way to override this?

**Comment by Char2sGu** [comment]
Any updates on this?

**Comment by Char2sGu** [comment]
Are there any problems if Akita is using a tslib version other than 2.4.1? If none, why use a specific version instead of a version range?

**Comment by hassanasad** [comment]
Facing same issue

**Comment by hassanasad** [comment]
@NetanelBasal Can you please have a look at this? :)

**Comment by Igor2017** [comment]
> @NetanelBasal Can you please have a look at this? :)

Need at your "package.json" write
{
     . . .
     "dependencies": {
         . . .
         "tslib": "2.4.1",
         . . .
}

**Comment by hassanasad** [comment]
@Igor2017 Yes but it would be better if its not fixed version. Probably `^2.4.1` would be better right?

**Comment by Igor2017** [comment]
> @Igor2017 Yes but it would be better if its not fixed version. Probably `^2.4.1` would be better right?

No. We need the exact version.

**Comment by sithwarrior** [comment]
> > @Igor2017 Yes but it would be better if its not fixed version. Probably `^2.4.1` would be better right?
> 
> No. We need the exact version.

Why do you need that exact version?

Facing the same issue when updating Akita to latest

**Comment by juancarlos-eco** [comment]
So it seems like Akita is totally abandoned...

**Comment by Char2sGu** [comment]
They now recommend @ngneat/elf in replacement of Akita, which is a much more composable solution comparing to Akita. It might seem pretty ugly at the first glance because it's more low-level comparing to Akita, but you can easily make Elf as powerful as Akita with just a little wrapping upon it.

**Comment by juancarlos-eco** [comment]
> They now recommend @ngneat/elf in replacement of Akita, which is a much more composable solution comparing to Akita. It might seem pretty ugly at the first glance because it's more low-level comparing to Akita, but you can easily make Elf as powerful as Akita with just a little wrapping upon it.

Yeah, I'm using it already for new projects and despite of one or two things that I really miss out, it's working great. 

Anyway, I don't really think it makes sense to leave Akita die like this, without even having a proper support for newer Angular versions... specially taking into account it has had a very successful path and there are a LOT of production projects using it. 

Just saying that, from my pov, this is a really sad ending for such a great tool. I ain't sure if Elf is going to end the same way therefore I won't be able to recommend it to everyone the same way I did with Akita.

**Comment by hassanasad** [comment]
Its a great tool which I am still using - Felt like it started going downhill when it went under Salesforce.
Pretty much like everyone else, trying to plan to migrate to Elf now.
